libsel4arm-vmm/src/devices.c

```diff
@@ -227,7 +227,7 @@ void *map_emulated_device(vm_t *vm, const struct device *d)
     /* Map the frame to the VM */
     DMAP("Mapping emulated device ipa0x%x\n", (uint32_t)vm_addr);
 
-    seL4_CapRights_t rights = seL4_CanRead;
+    seL4_CapRights_t rights = seL4_NoRights;
```

Comment:
Are you happy for me to revert this?

Reply:
Why should an emulated device have Read Rights?

We've emulated the serial device for the zynqmp, and when there are read rights, emulating doesn't work.

Reply:
Creating an emulated device without read or write rights doesn't require any frames to be mapped into the vm and this can be implemented with:
```c
-    vgic->dist = map_emulated_device(vm, &dev_vgic_dist);
+    vgic->dist = malloc(dev_vgic_dist.size);
```